### ISSUE TITLE
[DF] [BUG] `IndexError` when intersecting tables on >2 columns

### ISSUE THREAD

**Issue by charlesbluca** [dask-contrib/dask-sql#584]
**What happened**:
When trying to intersect tables on more than 3 columns, we are getting an `IndexError`:

<details>

```python-traceback
IndexError                                Traceback (most recent call last)
Input In [4], in <cell line: 11>()
      8 c.create_table("df1", pdf)
      9 c.create_table("df2", pdf)
---> 11 c.sql("select * from df1 intersect select * from df2")

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/context.py:493, in Context.sql(self, sql, return_futures, dataframes, gpu, config_options)
    489         self.create_table(df_name, df, gpu=gpu)
    491 rel, select_fields, _ = self._get_ral(sql)
--> 493 dc = RelConverter.convert(rel, context=self)
    495 if rel.get_current_node_type() == "Explain":
    496     return dc

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rel/convert.py:62, in RelConverter.convert(cls, rel, context)
     58 plugin_instance = cls.get_plugin(node_type)
     59 logger.debug(
     60     f"Processing REL {rel} using {plugin_instance.__class__.__name__}..."
     61 )
---> 62 df = plugin_instance.convert(rel, context=context)
     63 logger.debug(f"Processed REL {rel} into {LoggableDataFrame(df)}")
     64 return df

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rel/logical/join.py:185, in DaskJoinPlugin.convert(self, rel, context)
    180 # 7. Last but not least we apply any filters by and-chaining together the filters
    181 if filter_condition:
    182     # This line is a bit of code duplication with RexCallPlugin - but I guess it is worth to keep it separate
    183     filter_condition = reduce(
    184         operator.and_,
--> 185         [
    186             RexConverter.convert(rel, rex, dc, context=context)
    187             for rex in filter_condition
    188         ],
    189     )
    190     logger.debug(f"Additionally applying filter {filter_condition}")
    191     df = filter_or_scalar(df, filter_condition)

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rel/logical/join.py:186, in <listcomp>(.0)
    180 # 7. Last but not least we apply any filters by and-chaining together the filters
    181 if filter_condition:
    182     # This line is a bit of code duplication with RexCallPlugin - but I guess it is worth to keep it separate
    183     filter_condition = reduce(
    184         operator.and_,
    185         [
--> 186             RexConverter.convert(rel, rex, dc, context=context)
    187             for rex in filter_condition
    188         ],
    189     )
    190     logger.debug(f"Additionally applying filter {filter_condition}")
    191     df = filter_or_scalar(df, filter_condition)

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/convert.py:73, in RexConverter.convert(cls, rel, rex, dc, context)
     65     raise NotImplementedError(
     66         f"No conversion for class {expr_type} available (yet)."
     67     )
     69 logger.debug(
     70     f"Processing REX {rex} using {plugin_instance.__class__.__name__}..."
     71 )
---> 73 df = plugin_instance.convert(rel, rex, dc, context=context)
     74 logger.debug(f"Processed REX {rex} into {LoggableDataFrame(df)}")
     75 return df

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/core/call.py:915, in RexCallPlugin.convert(self, rel, expr, dc, context)
    906 def convert(
    907     self,
    908     rel: "LogicalPlan",
   (...)
    913 
    914     # Prepare the operands by turning the RexNodes into python expressions
--> 915     operands = [
    916         RexConverter.convert(rel, o, dc, context=context)
    917         for o in expr.getOperands()
    918     ]
    920     # Now use the operator name in the mapping
    921     # TODO: obviously this needs to not be hardcoded but not sure of the best place to pull the value from currently???
    922     schema_name = "root"

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/core/call.py:916, in <listcomp>(.0)
    906 def convert(
    907     self,
    908     rel: "LogicalPlan",
   (...)
    913 
    914     # Prepare the operands by turning the RexNodes into python expressions
    915     operands = [
--> 916         RexConverter.convert(rel, o, dc, context=context)
    917         for o in expr.getOperands()
    918     ]
    920     # Now use the operator name in the mapping
    921     # TODO: obviously this needs to not be hardcoded but not sure of the best place to pull the value from currently???
    922     schema_name = "root"

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/convert.py:73, in RexConverter.convert(cls, rel, rex, dc, context)
     65     raise NotImplementedError(
     66         f"No conversion for class {expr_type} available (yet)."
     67     )
     69 logger.debug(
     70     f"Processing REX {rex} using {plugin_instance.__class__.__name__}..."
     71 )
---> 73 df = plugin_instance.convert(rel, rex, dc, context=context)
     74 logger.debug(f"Processed REX {rex} into {LoggableDataFrame(df)}")
     75 return df

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/core/call.py:915, in RexCallPlugin.convert(self, rel, expr, dc, context)
    906 def convert(
    907     self,
    908     rel: "LogicalPlan",
   (...)
    913 
    914     # Prepare the operands by turning the RexNodes into python expressions
--> 915     operands = [
    916         RexConverter.convert(rel, o, dc, context=context)
    917         for o in expr.getOperands()
    918     ]
    920     # Now use the operator name in the mapping
    921     # TODO: obviously this needs to not be hardcoded but not sure of the best place to pull the value from currently???
    922     schema_name = "root"

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/core/call.py:916, in <listcomp>(.0)
    906 def convert(
    907     self,
    908     rel: "LogicalPlan",
   (...)
    913 
    914     # Prepare the operands by turning the RexNodes into python expressions
    915     operands = [
--> 916         RexConverter.convert(rel, o, dc, context=context)
    917         for o in expr.getOperands()
    918     ]
    920     # Now use the operator name in the mapping
    921     # TODO: obviously this needs to not be hardcoded but not sure of the best place to pull the value from currently???
    922     schema_name = "root"

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/convert.py:73, in RexConverter.convert(cls, rel, rex, dc, context)
     65     raise NotImplementedError(
     66         f"No conversion for class {expr_type} available (yet)."
     67     )
     69 logger.debug(
     70     f"Processing REX {rex} using {plugin_instance.__class__.__name__}..."
     71 )
---> 73 df = plugin_instance.convert(rel, rex, dc, context=context)
     74 logger.debug(f"Processed REX {rex} into {LoggableDataFrame(df)}")
     75 return df

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/physical/rex/core/input_ref.py:34, in RexInputRefPlugin.convert(self, rel, rex, dc, context)
     32 # The column is references by index
     33 index = rex.getIndex()
---> 34 backend_column_name = cc.get_backend_by_frontend_index(index)
     35 return df[backend_column_name]

File /opt/conda/envs/dsql/lib/python3.9/site-packages/dask_sql-2022.1.0+95.g8726c2c-py3.9-linux-x86_64.egg/dask_sql/datacontainer.py:146, in ColumnContainer.get_backend_by_frontend_index(self, index)
    141 def get_backend_by_frontend_index(self, index: int) -> str:
    142     """
    143     Get back the dask column, which is referenced by the
    144     frontend (SQL) column with the given index.
    145     """
--> 146     frontend_column = self._frontend_columns[index]
    147     backend_column = self._frontend_backend_mapping[frontend_column]
    148     return backend_column

IndexError: list index out of range
```

</details>

**What you expected to happen**:
Would expect the intersect to succeed, which it does if we intersect on less than 3 columns using identical tables

**Minimal Complete Verifiable Example**:

```python
import pandas as pd
from dask_sql import Context

c = Context()

pdf = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9]})

c.create_table("df1", pdf)
c.create_table("df2", pdf)

c.sql("select * from df1 intersect select * from df2")
```

**Environment**:

- dask-sql version: latest of `datafusion-sql-planner` with:
  - #575
  - #578
  - #581
- Python version: 3.9
- Operating System: ubuntu
- Install method (conda, pip, source): source

**Comment by jdye64** [comment]
FYI - #581 no longer resolves this issue due to scope changes. This is still an issue.

**Comment by jdye64** [comment]
Closed with #599
